libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPAuthCodeActivity.kt

```diff
@@ -1,53 +1,118 @@
+
     companion object {
         private val TAG: String = IDPAuthCodeActivity::class.java.simpleName
     }
 
+    private lateinit var _webView: WebView
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         // Protects against screenshots.
         window.setFlags(LayoutParams.FLAG_SECURE, LayoutParams.FLAG_SECURE)
 
-        // Set theme
-        val isDarkTheme = SalesforceSDKManager.getInstance().isDarkTheme
-        setTheme(if (isDarkTheme) R.style.SalesforceSDK_Dark_Login else R.style.SalesforceSDK)
-        SalesforceSDKManager.getInstance().setViewNavigationVisibility(this)
-
-        // Set layout
-        setContentView(R.layout.sf__idp_auth_code)
-
-        // Setup web view
-        val webView = findViewById<WebView>(R.id.sf__webview)
-        val webSettings = webView.settings
-        webSettings.useWideViewPort = true
-        webSettings.layoutAlgorithm = LayoutAlgorithm.NORMAL
-
-        (SalesforceSDKManager.getInstance().idpManager as? IDPManager)?.let { idpManager ->
-            val spAppPackageName = idpManager.getSrcAppPackageName(intent)
-            idpManager.attachToActiveFlow(this, this, spAppPackageName)
-            // Handing intent to IDPManager
-            idpManager.onReceive(this@IDPAuthCodeActivity, intent)
-        } ?: run {
-            SalesforceSDKLogger.d(this::class.java.simpleName, "no idp manager to handle ${LogUtil.intentToString(intent)}")
+        // TODO fix theming
+//        // Set theme
+//        val isDarkTheme = SalesforceSDKManager.getInstance().isDarkTheme
+//        setTheme(if (isDarkTheme) R.style.SalesforceSDK_Dark_Login else R.style.SalesforceSDK)
+//        SalesforceSDKManager.getInstance().setViewNavigationVisibility(this)
+
+        setContent {
+            IDPAuthCodeScreen(intent = intent)
+        }
+    }
+
+    @Composable
+    fun IDPAuthCodeScreen(intent: android.content.Intent) {
+        val context = LocalContext.current
+        val idpManager = SalesforceSDKManager.getInstance().idpManager as? IDPManager
+        var isLoading by remember { mutableStateOf(true) }
+
+        Box(
+            modifier = Modifier.fillMaxSize(),
+            contentAlignment = Alignment.Center
+        ) {
+            if (isLoading) {
+                CircularProgressIndicator()
+            }
+
+            AndroidView(
+                factory = { ctx ->
+                    WebView(ctx).apply {
+                        _webView = this
+
+                        settings.apply {
+                            javaScriptEnabled = true
+                            useWideViewPort = true
+                            layoutAlgorithm = WebSettings.LayoutAlgorithm.NORMAL
+                        }
+
+                        // Set WebViewClient and pass `isLoading` state
+                        webViewClient = object : android.webkit.WebViewClient() {
```

Comment:
Showing/hiding spinner used to be done in IDPAuthCodeHelper but should really be in view code i.e. here
IDPAuthCodeHelper now creates a WebViewClient that is wrapped around this one